### DISCUSSION TITLE
Oneliner in "def init(self)".

### DISCUSSION THREAD

**Discussion by traderblakeq** [kernc/backtesting.py#754] [Q&A] [answered]
Dear Community,

I was wondering what I can do to make this "oneliner" work.

Im trying to make a bool variable "self.rsi14"

The issue is the ">" than 50 calculation. 

I assume it is something with the datastructure I dont fully understand.

(Line 15):

<img width="397" alt="image" src="https://user-images.githubusercontent.com/36839989/193036707-5be3ae60-dbf9-41ec-9204-1174c6ddbb05.png">

Gives me this output in terminal.

<img width="526" alt="image" src="https://user-images.githubusercontent.com/36839989/193038564-9cc07ff6-7286-4418-aaf1-c0f0e296edd4.png">

(I know I can do the > 50 in next(self) def, but I would like to do that calculation in init if possible)

Thank you for any input :).

NB: Im a noob, so this might be a dumb question.

**Comment by kernc** [comment, marked as answer]
I'd do it like:
```py
self.rsi14 = self.I(lambda: ta.rsi(close.s), 14) > 50)
# or
self.rsi14 = self.I(lambda *args: ta.rsi(*args) > 50, self.data.Close.s, 14)

**Reply by traderblakeq** [reply]
Thank you! - Worked perfectly, and I did learn by your answer!.
